### PR TITLE
Fix mac

### PR DESCRIPTION
**Pull request by potuz**
Fixes compilation on Mac OS X with newer ARM boards with clang. 

Adds CI workflows for Mac OS X and Windows.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,6 +12,44 @@ jobs:
...
+      - uses: actions/checkout@v4
+      
+      - name: Install MinGW
+        run: |
+          choco install mingw
+          echo "C:\tools\mingw64\bin" >> $GITHUB_PATH
+
+      - name: Set CC environment variable
+        run: |
+          echo "CC=gcc" >> $GITHUB_ENV
```
Comment (prestonvanloon) [inline .github/workflows/test.yml:28]: Do you actually need this? If so, please use the `env` field https://docs.github.com/en/actions/learn-github-actions/variables

File: src/Makefile [hunk 2]
```diff
@@ -67,7 +67,11 @@ $(libname): $(objx86) hashtree.pc
 	$(AR) rcs $(libname) $(objx86)
 endif
 
+ifeq ($(WIN),1)
+all: $(libname) test
+else
 all: $(libname) test bench
+endif
```
Comment (prestonvanloon) [inline src/Makefile:74]: Please add a comment in the README that benchmarks cannot be run on windows

File: src/sha256_armv8_neon_x4.S [hunk 3]
```diff
@@ -277,15 +277,44 @@ TQ7		    .req q22
                     round_padding   \F, \G, \H, \A, \B, \C, \D, \E
 .endm
  
+#ifdef __APPLE__
+.global _hashtree_sha256_neon_x4
+#else
 .global hashtree_sha256_neon_x4
+#endif
+#ifdef __APPLE__
+//.type   hashtree_sha256_neon_x4,%function
```
Comment (prestonvanloon) [inline src/sha256_armv8_neon_x4.S:286]: What's up with this?

Comment (potuz) [reply src/sha256_armv8_neon_x4.S:286]: This is that the .type instruction doesn't exists in GNU assembler for Mac OS

**Review by prestonvanloon — DISMISSED** [review]

**Review by arnetheduck — APPROVED** [review]
Neat! passes SSZ tests on our mac/arm64 runner

**Comment by matthewkeil** [comment]
Sweet!  Is working with our JS bindings on arm mac locally and in CI.  You 🪨 dude!!

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**